plugins/system_upgrade.py

```diff
@@ -598,6 +610,9 @@ def run_upgrade(self):
         # change the upgrade status (so we can detect crashed upgrades later)
         command = ''
         with self.state as state:
+            if self.opts.poweroff_after is not None:
+                state.poweroff_after = self.opts.poweroff_after
+
```

Comment:
This whole added block is probably unwanted if I am not mistaken. When `reboot` is used with the `--poweroff` parameter, the `poweroff_after` value is saved into the state and then reboot is called. After that the state is loaded from the file and the `upgrade` here is called having the value already set.

Reply:
Right, this can be taken out.